### PR TITLE
LPS-133073 Avoid fetching until queryParams are processed

### PR DESCRIPTION
**Pull request by javierdearcos**
I've tried this strategy to avoid fetching until the queryParams are processed.

I initialized the page, pageSize, and search params to null instead of default values and check for these values before fetching.

In the queryParams hook this parameters will be loaded and then we can fetch the information

### CODE REVIEW

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/pages/home/UserActivity.es.js [hunk 1]
```diff
@@ -59,9 +61,27 @@ export default withRouter(
...
+					filter: `creatorId eq ${creatorId}`,
+					page,
+					pageSize,
+					siteKey,
+				},
+			}
+		);
+
+		useEffect(() => {
+			if (!page || !pageSize) {
```
Comment (nhpatt) [inline modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/pages/home/UserActivity.es.js:77]: but what happens if they are no parameters?

**Review by nhpatt — CHANGES_REQUESTED** [review]
I think the bug had 2 parts, right? the other one was using history.push instead of refreshing the page.

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/pages/home/UserActivity.es.js (hunk 1 above)
Comment (javierdearcos) [reply modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/pages/home/UserActivity.es.js:77]: I had this doubt and I've checked that the queryParams hooks are always executed once to put the default value

**Comment by javierdearcos** [comment]
> I think the bug had 2 parts, right? the other one was using history.push instead of refreshing the page.

I missed this second part. I will check it

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/pages/home/UserActivity.es.js (hunk 1 above)
Comment (nhpatt) [reply modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/pages/home/UserActivity.es.js:77]: But what are the defaults for the queryParams?

Comment (javierdearcos) [reply modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/pages/home/UserActivity.es.js:77]: The hook is always called at least once and fall in the default used in the hook, 1 for page, 20 for pageSize, and '' for search

Comment (nhpatt) [reply modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/pages/home/UserActivity.es.js:77]: Ok, I see they get it from the useEffect that is called when queryParams has something.

**Comment by nhpatt** [comment]
it's missing the second part but forwarding for now...

**Comment by nhpatt** [comment]
ci:forward

**Pull request closed without merging**